activerecord/lib/active_record/result.rb

```diff
@@ -34,13 +34,13 @@ module ActiveRecord
   class Result
     include Enumerable
 
-    attr_reader :columns, :rows, :column_types
+    attr_reader :columns, :rows, :types
```

Comment:
I don't see a strong reason to rename the interface/instance variable to `types`. This can break something that's not covered by tests, e.g. the `initialize_copy` is now broken because it only copies `@column_type` but not `@types`.

Reply:
🤔 Good callout re: `initialize_copy`! I'll audit uses of the instance variable in this class.

I wouldn't expect the _instance variable_ to be part of this classes interface, though, and I did implement `column_types` (below) to maintain the classes interface.

My concern with simply reusing `@column_types` for the new third argument to `initialize` is that the _type_ of that instance variable has changed (from a dictionary of types correlated to columns by name to an array of types correlated to columns by position).

Reply:
`initialize_copy` appears to be the only place where `@column_types` was referenced — I made that change! 👍 ✅